backend/src/config.rs

```diff
@@ -1,16 +1,138 @@
-use cli::Config;
+use cli::{Config, LoggerConfig};
+use file_rotate::{suffix::AppendTimestamp, ContentLimit, FileRotate};
+use thiserror::Error;
 use tower_http::cors::CorsLayer;
+use tracing_appender::non_blocking::{NonBlocking, WorkerGuard};
+use tracing_subscriber::{filter::LevelFilter, prelude::*, util::SubscriberInitExt};
 
+#[allow(clippy::module_name_repetitions)]
 pub trait ConfigExt {
     /// Return either very permissive [`CORS`](`CorsLayer`) configuration
     /// or empty one based on `cors_allow_all` field
     fn get_cors_configuration(&self) -> CorsLayer;
 }
 
+pub trait LoggerExt {
+    /// Initialize logger.
+    ///
+    /// Returns [`WorkerGuard`]s for off-thread writers.
+    /// Should not be dropped.
+    ///
+    /// # Errors
+    ///
+    /// Function returns error if `init_file_rotate` fails
+    fn init_logger(&self) -> Result<Vec<WorkerGuard>, LoggerError>;
+
+    /// Returns [`std:io::Write`] object that rotates files on write
+    ///
+    /// # Errors
+    ///
+    /// Function returns error if `log_file` is not specified
+    fn init_file_rotate(&self) -> Result<FileRotate<AppendTimestamp>, LoggerError>;
+
+    /// Returns non-blocking file writer
+    ///
+    /// Also returns [`WorkerGuard`] for off-thread writing.
+    /// Should not be dropped.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the file logger configuration is empty, file logging
+    /// is disabled or logs filename is not specified
+    fn non_blocking_file_writer(&self) -> Result<(NonBlocking, WorkerGuard), LoggerError>;
+
+    /// Returns non-blocking stdout writer
+    ///
+    /// Also returns [`WorkerGuard`] for off-thread writing.
+    /// Should not be dropped.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the stdout logger configuration is empty or stdout logging
+    /// is disabled
+    fn non_blocking_stdout_writer(&self) -> Result<(NonBlocking, WorkerGuard), LoggerError>;
+}
+
 impl ConfigExt for Config {
     fn get_cors_configuration(&self) -> CorsLayer {
         self.cors_allow_all
             .then_some(CorsLayer::very_permissive())
             .unwrap_or_default()
     }
 }
+
+impl LoggerExt for LoggerConfig {
+    fn init_logger(&self) -> Result<Vec<WorkerGuard>, LoggerError> {
+        let mut guards = Vec::with_capacity(2);
+
+        let mut layers_iter = [
+            self.non_blocking_file_writer().ok(),
+            self.non_blocking_stdout_writer().ok(),
```

Comment:
This is bad, because all errors are swallowed here. You have, for example, `LoggerError::NoFileName` error, which is better to propagate above. I suggest to check right here, that the particular config is `Some()` and Writer is `enabled` (by the flag), and only if those checks pass, then create the Writer. Any error in this case can be propagated above with `?` operator